_studio/shared/umc/codec/jpeg_dec/src/umc_mjpeg_mfx_decode_hw.cpp

```diff
@@ -398,6 +398,13 @@ Status MJPEGVideoDecoderMFX_HW::GetFrameHW(MediaDataEx* in)
             foundSOS = true;
         }
 
+        CMemBuffInput* stream = new CMemBuffInput();
+        stream->Open((uint8_t*)in->GetDataPointer() + pAuxData->offsets[i] + 2, (int32_t)in->GetDataSize() - pAuxData->offsets[i] - 2);
+
+        jerr = m_decBase->SetSource(stream);
+        if(JPEG_OK != jerr)
+            return UMC_ERR_FAILED;
+
```

Comment:
This change looks code refactoring, please (removal of code duplication), extract it to a separate change!